### PR TITLE
Blog app - creating a data model

### PR DESCRIPTION
**Pull request by EidHachem**
## In this PR I implemented the below:

- Created a new app called `gem blogs`.
- set up PostgreSQL as the default database.
- Created project schema by creating the below tables:
    - User
    - Post
    - Comment
    - Like
-  Added foreign keys and indexes.

### CODE REVIEW

File: db/migrate/20220809130247_add_author_id_ref_to_posts.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddAuthorIdRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
```
Comment (medaminedev66) [inline db/migrate/20220809130247_add_author_id_ref_to_posts.rb]: - [ ] The reference is not directly to user, it is instead author to table users. Kindly see the below example on how to make such migration for adding a reference with a different name than the reference model class name and make changes in your database. 🙏

image

File: db/migrate/20220809130303_add_author_id_ref_to_comments.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddAuthorIdRefToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
```
Comment (medaminedev66) [inline db/migrate/20220809130303_add_author_id_ref_to_comments.rb]: - [ ] Again, the reference name should be `author`. Please refer to the above similar comment for the other migration.

File: db/migrate/20220809130316_add_author_id_ref_to_likes.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddAuthorIdRefToLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, null: false, foreign_key: true
```
Comment (medaminedev66) [inline db/migrate/20220809130316_add_author_id_ref_to_likes.rb]: - [ ] Again, the reference name should be `author`. Please refer to the above similar comment for the other migration.

File: db/schema.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    t.integer "post_counter"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+  add_foreign_key "comments", "posts"
+  add_foreign_key "comments", "users"
+  add_foreign_key "likes", "posts"
+  add_foreign_key "likes", "users"
+  add_foreign_key "posts", "users"
```
Comment (medaminedev66) [inline db/schema.rb]: After applying the requested changes in the migrations, this should look like this:

image

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @EidHachem  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- There are no linter errors
- You have a descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20220809130247_add_author_id_ref_to_posts.rb (hunk 1 above)
Comment (EidHachem) [reply db/migrate/20220809130247_add_author_id_ref_to_posts.rb]: Hi, @medaminedev66 Thank you for the review, I was wondering if I can update it directly in the file and then run db:migrate again or should I delete the database and create it again?

**Review by Davidosky007 — APPROVED** [review]
Hi @EidHachem 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well wriiten PR description 👍 
* Proper use of Gitflow 👍 
## Optional suggestions
## I strongly suggesst you add a descriptive readme to your project this would contain infomation about what your app is all about. [Kindly customise this readme template to suit your project](https://github.com/microverseinc/readme-template)
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by EidHachem** [comment]
@Davidosky007 Thank you so much, I will update the readme file now

File: db/migrate/20220809130247_add_author_id_ref_to_posts.rb (hunk 1 above)
Comment (medaminedev66) [reply db/migrate/20220809130247_add_author_id_ref_to_posts.rb]: Hi @EidHachem. Yes, you can update the migration directly, but do not forget to reset the database by running `rails db:reset`, then run `rails db:migrate`.

**Pull request merged**